src/thing/remove.ts

```diff
@@ -169,6 +170,28 @@ export const removeDatetime: RemoveOfType<Date> = (thing, property, value) => {
   );
 };
 
+/**
+ * Create a new Thing with the given date removed for the given Property.
+ *
+ * The original `thing` is not modified; this function returns a cloned Thing with updated values.
+ *
+ * @param thing Thing to remove a date value from.
+ * @param property Property for which to remove the given date value.
+ * @param value Date to remove from `thing` for the given `property`.
+ * @returns A new Thing equal to the input Thing with the given value removed for the given Property.
+ */
+export const removeDate: RemoveOfType<Date> = (thing, property, value) => {
+  internal_throwIfNotThing(thing);
+  // deserialize date will return Date object at midday (43200000 ms)
+  return removeLiteralMatching(
+    thing,
+    property,
+    xmlSchemaTypes.date,
+    (foundDate) =>
+      deserializeDate(foundDate)?.getTime() === (value.getTime() + 43200000)
```

Comment:
Hmm.... So this line gets me thinking: if the developer has to pass in a `Date` object not only with the correct date, but also with a certain time - how much value is it providing that we're using a `Date` object here?

Would it potentially make more sense to use a custom plain object here using only the properties we're interested in? So something like this:

```
type Date = {
  year: number;
  month: number;
  day: number;
  timezoneHourOffset?: number;
  timezoneMinuteOffset?: number;
};
```

Alternatively, if that's cumbersome too, maybe we should only check on `foundDate`'s and `value`'s year, month and day rather than adding 43200000ms?

Sorry, I'm being a little annoying here bring this up after implementing it one way, but it's the implementation experience that brought the insight...

Reply:
Yeah, it's a fair point. On the set/get/add end, the user can pass in any date object and the serializing will handle all of the annoying date-related things. So, for the user, it's still fairly simple/easy. For example, the following code will add "1990-11-12Z" to the thing, ignoring the time :

```
const updatedThing = addDate(     
 thing,      
"https://some.vocab/predicate", 
 new Date(Date.UTC(1990, 10, 12, 3, 3, 3))
);
```
The difficulty, as you say, is on the remove side. In this case, they would have to provide the full date object with the time set to 0 in order to get a match, not ideal or intuitive.

I'm reluctant to completely ditch the Date object because I like how smoothly it works for appending to things, but for the remove method, we need a better solution. In my opinion, I think modifying the code to check for a date match should be enough. In that use case, a user just has to match the date they set originally which I think is pretty okay from a UX point of view, and, we can remove the clunky 43200000ms...

Reply:
Also, thanks for git tips, useful stuff to know for the future.

Reply:
OK, sounds good!

And how about time: would a custom object make sense there, since it's somewhat more clunky to construct a `Date` just for the time?

Reply:
Yeah, I think that's a good idea. Returning the user 1970 Jan 1st attached to their time isn't nice. Something like this...

```
type Time = {
  hour: number;
  minute: number;
  second: number;
  timezoneHourOffset?: number;
  timezoneMinuteOffset?: number;
};
```

We'd need to document this change well, is that something I need to do before the PR is merged?

Reply:
You're still working on time right? So it needn't block this PR.

For the time PR/branch, you can define that type and then export it from `index.ts` and add a TypeDoc comment block of documentation to that type as well.

Reply:
Yeah oops for the other PR. Okay, I'll do that.